apple/Sources/FerrostarMapLibreUI/Views/InstructionsView.swift

```diff
@@ -0,0 +1,121 @@
+import FerrostarCoreFFI
+import SwiftUI
+
+/// <#Description#>
+struct InstructionsView: View {
+    private let visualInstruction: VisualInstruction
+    private let primaryRowTheme: InstructionRowTheme
+    private let secondaryRowTheme: InstructionRowTheme
+    private var hasSecondary: Bool {
+        visualInstruction.secondaryContent != nil
+    }
+
+    /// Create a visual instruction banner view. This view automatically displays the secondary
+    /// instruction if there is one.
+    ///
+    /// - Parameters:
+    ///   - visualInstruction: The visual instruction to display.
+    ///   - primaryRowTheme: The theme for the primary instruction.
+    ///   - secondaryRowTheme: The theme for the secondary instruction.
+    public init(
+        visualInstruction: VisualInstruction,
+        primaryRowTheme: InstructionRowTheme = DefaultInstructionRowTheme(),
+        secondaryRowTheme: InstructionRowTheme = DefaultSecondaryInstructionRowTheme()
+    ) {
+        self.visualInstruction = visualInstruction
+        self.primaryRowTheme = primaryRowTheme
+        self.secondaryRowTheme = secondaryRowTheme
+    }
+
+    var body: some View {
+        VStack {
+            DefaultManeuverInstructionView(
+                text: visualInstruction.primaryContent.text,
+                maneuverType: visualInstruction.primaryContent.maneuverType,
+                maneuverModifier: .left,
+                distanceRemaining: "15 km", // TODO: Dynamic distance to step
+                theme: primaryRowTheme
+            )
+            .font(.title2.bold())
+            .padding(.horizontal, 16)
+            .padding(.top, 16)
+            .padding(.bottom, 0)
+
+            if let secondaryContent = visualInstruction.secondaryContent {
+                VStack {
+                    DefaultManeuverInstructionView(
+                        text: secondaryContent.text,
+                        maneuverType: secondaryContent.maneuverType,
+                        maneuverModifier: secondaryContent.maneuverModifier,
+                        distanceRemaining: "500 m", // TODO: Dynamic distance to step
```

Comment:
We likely need distance remaining on current instruction as well & full distance on visual instruction for anything after current/generally.